library/cc/stream_prototype.h

```diff
@@ -6,31 +6,30 @@
 #include <functional>
 #include <memory>
 
-#include "engine.h"
 #include "envoy_error.h"
 #include "executor.h"
+#include "library/common/types/c_types.h"
 #include "response_headers.h"
 #include "response_trailers.h"
 #include "stream.h"
 #include "stream_callbacks.h"
 
-class Engine;
-
 class StreamPrototype {
 public:
-  StreamPrototype(Engine engine);
+  StreamPrototype(envoy_engine_t engine);
```

Comment:
here and in `StreamClient` i ended up passing down an `envoy_engine_t` because it fixes a whole bunch of `#include` cycles. thoughts?